src/Entity/TaskListRepository.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use AppBundle\Api\Dto\MyTaskList;
+use AppBundle\Api\Dto\MyTaskDto;
+use AppBundle\Api\Dto\TaskMetadataDto;
+use AppBundle\Api\Dto\TaskPackageDto;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class TaskListRepository extends ServiceEntityRepository
+{
+    public function __construct(
+        ManagerRegistry $registry,
+        private readonly EntityManagerInterface $entityManager
+    )
+    {
+        parent::__construct($registry, TaskList::class);
+
+    }
+
+    public function findMyTaskListAsDto(UserInterface $user, \DateTime $date): ?MyTaskList
+    {
+        $taskList = $this->findOneBy([
+            'courier' => $user,
+            'date' => $date,
+        ]);
+
+        if (null === $taskList) {
+            return null;
+        }
+
+        $taskIds = $taskList->getItems()->map(function (TaskList\Item $item) {
+            return $item->getTask()->getId();
+        });
+
+        $queryResult = $this->entityManager->createQueryBuilder()
+            ->select([
+                't',
+                'o.number AS orderNumber',
+                'o.total AS orderTotal',
+                'org.name AS organizationName',
+                // objects are listed below to force them being hydrated / pre-fetched by Doctrine
+                // https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/reference/dql-doctrine-query-language.html#result-format
+                'taskPackage',
+                'package',
+                'incidents',
+            ])
+            ->from(Task::class, 't')
+            ->leftJoin('t.delivery', 'd')
+            ->leftJoin('d.order', 'o')
+            ->leftJoin('t.organization', 'org')
+            ->leftJoin('t.packages', 'taskPackage')
+            ->leftJoin('taskPackage.package', 'package')
+            ->leftJoin('t.incidents', 'incidents')
+            ->where('t.id IN (:taskIds)')
+            ->setParameter('taskIds', $taskIds)
+            ->getQuery()
+            ->getResult();
+
+        //TODO; sort by position
+
+        $tasks = array_map(function ($row) {
+            $task = $row[0];
+            $taskDto = new MyTaskDto(
+                $task->getId(),
+                $task->getCreatedAt(),
+                $task->getUpdatedAt(),
+                $task->getType(),
+                $task->getStatus(),
+                $task->getAddress(),
+                $task->getDoneAfter(),
+                $task->getDoneBefore(),
+                $task->getPrevious()?->getId(),
+                $task->getNext()?->getId(),
+                $task->getTags(),
+                $task->isDoorstep(),
+                $task->getComments(),
+                $task->getPackages()->map(function (Task\Package $package) {
```

Comment:
it is wrong here `task.packages` depends on whether the task is a pickup or a dropoff
if pickup then the packages are the "sum" of the drop packages
see here https://github.com/coopcycle/coopcycle-web/blob/master/src/Action/Task/Collection.php#L11